stable/grafana/templates/deployment.yaml

```diff
@@ -1,123 +1,134 @@
-apiVersion: extensions/v1beta1
+apiVersion: apps/v1beta2
 kind: Deployment
 metadata:
+  name: {{ template "grafana.fullname" . }}
   labels:
-    app: {{ template "grafana.fullname" . }}
-    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
-    component: "{{ .Values.server.name }}"
-    heritage: "{{ .Release.Service }}"
-    release: "{{ .Release.Name }}"
-  name: {{ template "grafana.server.fullname" . }}
+    app: {{ template "grafana.name" . }}
+    chart: {{ template "grafana.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+{{- with .Values.annotations }}
+  annotations:
+{{ toYaml . | indent 4 }}
+{{- end }}
 spec:
-  replicas: {{ .Values.server.replicaCount }}
+  replicas: {{ .Values.replicas }}
+  selector:
+    matchLabels:
+      app: {{ template "grafana.name" . }}
+      release: {{ .Release.Name }}
   template:
     metadata:
-      annotations:
-        checksum/config: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum }}
-        checksum/dashboards-config: {{ include (print $.Template.BasePath "/dashboards-configmap.yaml") . | sha256sum }}
-      {{- range $key, $value := .Values.server.annotations }}
-        {{ $key }}: {{ $value | quote }}
-      {{- end }}
       labels:
-        app: {{ template "grafana.fullname" . }}
-        component: "{{ .Values.server.name }}"
-        release: "{{ .Release.Name }}"
+        app: {{ template "grafana.name" . }}
+        release: {{ .Release.Name }}
     spec:
-      nodeSelector:
-{{ toYaml .Values.server.nodeSelector | indent 8 }}
-  {{- if .Values.server.tolerations }}
-      tolerations:
-{{ toYaml .Values.server.tolerations | indent 8 }}
-  {{- end }}
-      {{- if .Values.server.antiAffinity.enabled }}
-      affinity:
-        podAntiAffinity:
-          {{- $antiAffinityType := default "hard" .Values.server.antiAffinity.type | lower }}
-          {{- if eq $antiAffinityType "hard" }}
-          requiredDuringSchedulingIgnoredDuringExecution:
-          - topologyKey: "kubernetes.io/hostname"
-            labelSelector:
-              matchLabels:
-                app: {{ template "grafana.server.fullname" . }}
-          {{- end }}
-          {{- if eq $antiAffinityType "soft" }}
-          preferredDuringSchedulingIgnoredDuringExecution:
-          - weight: 1
-            podAffinityTerm:
-              topologyKey: "kubernetes.io/hostname"
-              labelSelector:
-                matchLabels:
-                  app: {{ template "grafana.server.fullname" . }}
-          {{- end }}
-      {{- end }}
+{{- if .Values.dashboards }}
       initContainers:
-        - name: copy-configs
-          image: busybox
-          command: ['sh', '-c', 'cp /tmp/config-volume-configmap/* /tmp/config-volume 2>/dev/null || true; cp /tmp/dashboard-volume-configmap/* /tmp/dashboard-volume 2>/dev/null || true']
+        - name: download-dashboards
+          image: appropriate/curl
```

Comment:
Could this also be a setting in values.yaml? Use case there: using a private mirror for deployment

Reply:
@DaytonG Sure.  I can make the download-dashboard image a setting in values.yaml

Reply:
Re: the init container: It was my intention to have it only run when there were dashboards to download, but I haven't been able to figure out how.  I will restructure .Values.dashboards to hopefully make it a bit easier.
Any help would be appreciated!

Reply:
Hm. The simplest change could be to just move the downloaded dashboards to its own variable and process it independently of the dashboards provided locally. Could potentially increase configuration complexity a little, but it may also simplify some of the template code.

Reply:
I thought about that, but I'd like to avoid the configuration complexity.  I'll ruminate on it for a while.  Even if the initContainer runs unnecessarily, the image is light (~7mb) and with nothing to download the script is benign.